### DISCUSSION TITLE
Further extension of Umbraco Forms states

### DISCUSSION THREAD

**Discussion by CDSFounder** [umbraco/Umbraco.Forms.Issues#1522] [Ideas]
We were very happy to see a new "Rejected" state added to Umbraco Forms in a [recent release.](https://github.com/umbraco/Umbraco.Forms.Issues/issues/716). 

We are requesting it be possible to extend this to allow for configurable or extendable form states. We would like to have more states than just Approve, Reject, Submitted.

As an example: We would like to have a form state "Re-opened" where a form is technically rejected, but the state Re-opened kicks it back to the member's front-end portal with a note for why it was rejected. The member could then see a list of their submitted forms and "continue" the form listed as "Re-opened" along with the note for why it was re-opened and proceed to fix any issues and re-submit the form. This is different than a rejected state because the member would be allowed to resubmit the form with minor corrections, as opposed to simply permanently rejecting what was submitted.

As another example: In the case of multi-page forms, it would be nice to allow the member to save the form progress and come back to it later. In this case, the form would be in an "In Progress" state.

If extending Umbraco forms for dynamic / configurable states is too bug if a lift, it would be nice to statically have a Re-opened and In Progress state. :)

**Comment by marcoteodoro** [comment]
Hello everyone, 

As we need to implement the status check against the reports soon, we would appreciate your input on the preferred approach. We want to avoid reinventing the wheel and make the process as efficient as possible. 

As mentioned by @CDSFounder, we would like to have the ability to change the status of the reports to "Re-opened". However, the only way to achieve this currently is to create a custom AngularJS control and manage the information on a custom table using the ReportId/Statuses. We want to minimize making changes on the Umbraco back office for now, especially since AngularJS is coming to an end soon. 

Could you please let us know if you think this implementation will be available soon or if it will take quite some time to be available for us? 

Thank you.

**Comment by AndyButland** [comment]
We don't have any current plans to add this @marcoteodoro.  As you'll appreciate the feature request was only made yesterday, so it's not something we have considered or have on a road map.  It's a nice idea and maybe something will come from it in the core product in future, but if you need the feature and have a plan for implementing it yourselves I'd suggest to go ahead with that.

**Comment by marcoteodoro** [comment]
Thanks for the quick response @AndyButland! We need to address the issue, but since AngularJS is near its end, I am considering waiting until Bellissima is released. Our plan is to upgrade our solution to version 14 first and then implement custom Umbraco forms, entry list view, and additional forms entry editor. 

But we may need to do it earlier...

I checked the roadmap at https://umbraco.com/products/knowledge-center/roadmap/, and it mentions improvements for Umbraco Forms, such as adding a progression display to the default theme and an optional final confirmation page for review or editing of the pending form submission.

At this point, you may want to consider allowing additional statuses. One way to do this is by having the admin set the status to "Re-opened," so the member who submitted the form can easily open the form on the review page. This is just an idea, though. 

Do you have any thoughts or suggestions on whether we should wait or proceed now? If we proceed, what would be the most effective way to ensure that we can reuse most of it once version 14 is out?

**Comment by TvGessel** [comment]
Since this is an older discussion and there are no current plans to extend the available states, I'm going to close this. Hopefully, you've found a suitable solution in the meantime.
